data-structures-queues/take-value-at-index.js

```diff
@@ -0,0 +1,10 @@
+/* exported takeValueAtIndex */
+
+function takeValueAtIndex(queue, index) {
+  for (let i = 0; i < index; i++) {
+    queue.enqueue(queue.peek());
+    queue.dequeue();
+  }
+
+  return queue.dequeue();
+}
```

Comment:
If the `queue` is empty, `takeValueAtIndex` should still return `undefined` when `index` is greater than `0`.